### PR TITLE
Bootstrap bump

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
Bumps bootstrap compiler to just-released beta.

https://forge.rust-lang.org/release/process.html#master-bootstrap-update-t-2-day-tuesday

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
rustfmt changes seem potentially more widespread than I'd expect, so asked on Zulip about them - https://rust-lang.zulipchat.com/#narrow/stream/266220-t-rustdoc/topic/expected.20changes.20in.20bootstrap.20bump.3F

cc @rust-lang/rustdoc as well

**Comment by Mark-Simulacrum** [comment]
er, @rust-lang/rustfmt, sorry

**Comment by calebcartwright** [comment]
> rustfmt changes seem potentially more widespread than I'd expect, so asked on Zulip about them - https://rust-lang.zulipchat.com/#narrow/stream/266220-t-rustdoc/topic/expected.20changes.20in.20bootstrap.20bump.3F
> 
> cc @rust-lang/rustdoc as well

Acknowledge seeing this, from a quick glance many of the changes look related to things like the newly added support for formatting let chains (some of the diffs are to lines contained within the body of a let chain). Can take a deeper look later though

**Comment by bjorn3** [comment]
cg_clif's build system has to respect `RUSTFLAGS` to pick up the right linker and such. I can add `-Zcheck-cfg=` to it though.

**Comment by Urgau** [comment]
> cg_clif's build system has to respect `RUSTFLAGS` to pick up the right linker and such. I can add `-Zcheck-cfg=` to it though.

Adding `--check-cfg=cfg(no_unstable_features)` to every `rustc` invocation for `mini_core` should do it.

**Comment by bjorn3** [comment]
Adding that here should work: https://github.com/rust-lang/rust/blob/d97bb195bf7b84815622b235cb51171c1b8f7e51/compiler/rustc_codegen_cranelift/build_system/tests.rs#L459

**Comment by Mark-Simulacrum** [comment]
Alright, looks like CI is passing now. This will probably continue to have merge conflicts / CI failures due to rustfmt formatting more code now, but can rebase as needed once review is done.

**Review by clubby789 — APPROVED** [review]
r=me

File: compiler/rustc_hir_typeck/src/expr.rs
Comment (ytmimi) [inline compiler/rustc_hir_typeck/src/expr.rs:1]: I think these changes are related to support for formatting match arms with `let-guards`

File: compiler/rustc_hir_analysis/src/astconv/errors.rs [hunk 1]
```diff
@@ -206,15 +206,14 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
...
+                    if generics.bounds_for_param(def_id).flat_map(|pred| pred.bounds.iter()).any(
+                        |b| match b {
                             hir::GenericBound::Trait(t, ..) => {
                                 t.trait_ref.trait_def_id().as_ref() == Some(best_trait)
                             }
                             _ => false,
-                        })
-                    {
+                        },
+                    ) {
```
Comment (ytmimi) [inline compiler/rustc_hir_analysis/src/astconv/errors.rs:216]: Now that we're formatting the `let-chain` above we're able to format the body this block and the `else-if` block below.

File: compiler/rustc_hir_analysis/src/astconv/bounds.rs [hunk 2]
```diff
@@ -194,8 +194,7 @@ impl<'tcx> dyn AstConv<'tcx> + '_ {
 
         self.add_bounds(
             param_ty,
-            ast_bounds.iter().filter(|bound| {
-                match filter {
+            ast_bounds.iter().filter(|bound| match filter {
```
Comment (ytmimi) [inline compiler/rustc_hir_analysis/src/astconv/bounds.rs:197]: One of the match arms below includes a `let-chain` in it's body. I believe that rustfmt failed to format that match arm before support for `let-chains` was added resulting in a failure to format the entire match expressions.

File: compiler/rustc_expand/src/config.rs [hunk 3]
```diff
@@ -251,8 +253,7 @@ impl<'a> StripUnconfigured<'a> {
         let trees: Vec<_> = stream
             .0
             .iter()
-            .flat_map(|tree| {
-                match tree.clone() {
+            .flat_map(|tree| match tree.clone() {
```
Comment (ytmimi) [inline compiler/rustc_expand/src/config.rs:256]: I believe these changes are related to support for formatting `let-guards`

File: compiler/rustc_const_eval/src/interpret/operator.rs [hunk 4]
```diff
@@ -114,9 +114,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         use rustc_middle::mir::BinOp::*;
 
         // Performs appropriate non-deterministic adjustments of NaN results.
-        let adjust_nan = |f: F| -> F {
-            if f.is_nan() { M::generate_nan(self, &[l, r]) } else { f }
-        };
+        let adjust_nan =
+            |f: F| -> F { if f.is_nan() { M::generate_nan(self, &[l, r]) } else { f } };
```
Comment (ytmimi) [inline compiler/rustc_const_eval/src/interpret/operator.rs:118]: These formatting changes don't seem related to `let-else`, `let-chain` or `let-gurad` support.

File: compiler/rustc_ast_passes/src/ast_validation.rs [hunk 5]
```diff
@@ -1213,8 +1213,12 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
...
                             errors::TildeConstReason::Function { ident: ident.span }
                         }
-                        &DisallowTildeConstContext::Trait(span) => errors::TildeConstReason::Trait { span },
-                        &DisallowTildeConstContext::Impl(span) => errors::TildeConstReason::Impl { span },
+                        &DisallowTildeConstContext::Trait(span) => {
+                            errors::TildeConstReason::Trait { span }
+                        }
+                        &DisallowTildeConstContext::Impl(span) => {
+                            errors::TildeConstReason::Impl { span }
+                        }
```
Comment (ytmimi) [inline compiler/rustc_ast_passes/src/ast_validation.rs:1221]: I believe these change are related to recent support for formatting match arms with `let guards`.

**Review by ytmimi — COMMENTED** [review]
Just went through all the files. The large majority of formatting changes seem related to `let-chain` support in rustfmt. There were also a handful that seemed related to `let-guards` in match arms. I tried to give explanations for any of the non-obvious changes as well as call out a few formatting changes that I thought were unexpected.

**Comment by chenrui333** [comment]
- for what is worth, it has fixed #117299
